lib/reference/properties.mjs

```diff
@@ -58,6 +58,11 @@ export const longhandSubPropertiesOfShorthandProperties = new Map([
 			'border-right-color',
 			'border-bottom-color',
 			'border-left-color',
+			'border-color',
+			'border-style',
+			'border-block',
+			'border-block-end',
+			'border-block-start',
```

Comment:
Right. This `longhandSubPropertiesOfShorthandProperties` is exposed under the `reference` type, so we need to keep its correctness.

https://github.com/stylelint/stylelint/blob/a5965a09b557fd4a5420a6b48f1c1353ecf83aa2/types/stylelint/index.d.ts#L826-L831

Reply:
So, I guess some specific logic for this rule is necessary.

---

Also, supporting `border-image` seems reasonable to me because the [CSS spec](https://drafts.csswg.org/css-backgrounds/#propdef-border) says:

> The [border](https://drafts.csswg.org/css-backgrounds/#propdef-border) shorthand also resets [border-image](https://drafts.csswg.org/css-backgrounds/#propdef-border-image) to its initial value.

E.g.

```css
a {
  border-image: url('...'); /* this is ignored by border */
  border: 1px solid black; /* here, border-image is reset */
}
```

Reply:
> So, I guess some specific logic for this rule is necessary.

Sorry, this comment was inaccurate. I mean that the following 3 properties addition makes sense:

```js
'border-width',
'border-style',
'border-color',
```

This order reflects the `border` shorthand's [values order](https://developer.mozilla.org/en-US/docs/Web/CSS/border#values).

But `border-image` cannot be added here because it's for resetting a value, unlike other properties. If we'd like to support `border-image`, we should write custom logic in `declaration-block-no-shorthand-property-overrides/index.mjs`.

Reply:
If supporting `border-image` is difficult, I think only `border-width`, `border-style`, and `border-color` are enough for this PR. 👌🏼

Reply:
> I think only border-width, border-style, and border-color are enough for this PR.

👍 

> The border shorthand also resets border-image to its initial value.

@ybiquitous can you create a separate issue?

Reply:
Sure. Just opened #7586.